optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -0,0 +1,423 @@
+                Provider option dictionary corresponding to the provider used. See available options
+                for each provider: https://onnxruntime.ai/docs/api/c/group___global.html . Defaults to `None`.
+        """
+        vae_decoder_session = ORTModel.load_model(vae_decoder_path, provider, session_options, provider_options)
+        text_encoder_session = ORTModel.load_model(text_encoder_path, provider, session_options, provider_options)
+        unet_session = ORTModel.load_model(unet_path, provider, session_options, provider_options)
+
+        return vae_decoder_session, text_encoder_session, unet_session
+
+    def _save_pretrained(
+        self,
+        save_directory: Union[str, Path],
+        vae_decoder_file_name: str = ONNX_WEIGHTS_NAME,
+        text_encoder_file_name: str = ONNX_WEIGHTS_NAME,
+        unet_file_name: str = ONNX_WEIGHTS_NAME,
+        **kwargs,
+    ):
+        save_directory = Path(save_directory)
+        src_to_dst_path = {
+            self.vae_decoder_model_path: save_directory
+            / DIFFUSION_MODEL_VAE_DECODER_SUBFOLDER
+            / vae_decoder_file_name,
+            self.text_encoder_model_path: save_directory
+            / DIFFUSION_MODEL_TEXT_ENCODER_SUBFOLDER
+            / text_encoder_file_name,
+            self.unet_model_path: save_directory / DIFFUSION_MODEL_UNET_SUBFOLDER / unet_file_name,
+        }
+
+        # TODO: Modify _get_external_data_paths to give dictionnary
+        src_paths = list(src_to_dst_path.keys())
+        dst_paths = list(src_to_dst_path.values())
+        # Add external data paths in case of large models
+        src_paths, dst_paths = _get_external_data_paths(src_paths, dst_paths)
+
+        for src_path, dst_path in zip(src_paths, dst_paths):
+            dst_path.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copyfile(src_path, dst_path)
+
+        self.tokenizer.save_pretrained(save_directory.joinpath("tokenizer"))
+        self.scheduler.save_pretrained(save_directory.joinpath("scheduler"))
+        if self.feature_extractor is not None:
+            self.feature_extractor.save_pretrained(save_directory.joinpath("feature_extractor"))
+
+    @classmethod
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        config: Dict[str, Any],
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        cache_dir: Optional[str] = None,
+        vae_decoder_file_name: str = ONNX_WEIGHTS_NAME,
+        text_encoder_file_name: str = ONNX_WEIGHTS_NAME,
+        unet_file_name: str = ONNX_WEIGHTS_NAME,
+        local_files_only: bool = False,
+        provider: str = "CPUExecutionProvider",
+        session_options: Optional[ort.SessionOptions] = None,
+        provider_options: Optional[Dict[str, Any]] = None,
+        use_io_binding: Optional[bool] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        **kwargs,
```

Comment:
Why?

Reply:
Argument `trust_remote_code` and `force_download` are not used (should be corrected in `OptimizedModel.from_pretrained` in an other PR)

Reply:
Then add explicitly `trsut_remote_code` and `force_download`, using `kwargs` when absolutely not needed is quite bug-prone.